warp10/src/main/java/io/warp10/continuum/store/MultiScanGTSDecoderIterator.java

```diff
@@ -208,42 +217,19 @@ public boolean hasNext() {
     bb.putLong(metadata.getClassId());
     bb.putLong(metadata.getLabelsId());
     
-    //
-    // We need to stop on the modulus boundary that precedes the last valid boundary
-    //
-    
-    boolean endAtMinlong = false;
-    
-    if (timespan >= 0) {
-      try {
-        modulus = Math.subtractExact(now, timespan);
-      } catch (ArithmeticException ae) {
-        modulus = Long.MIN_VALUE;
-        endAtMinlong = true;
-      }
-      
-      if (modulus < Long.MIN_VALUE + Constants.DEFAULT_MODULUS) {
-        endAtMinlong = true;
-        modulus = Long.MIN_VALUE;
-      } else {
-        modulus = (modulus - (modulus % Constants.DEFAULT_MODULUS)) - Constants.DEFAULT_MODULUS;
-      }
-    
-      bb.putLong(Long.MAX_VALUE - modulus);
-    } else {
-      // Long.MAX_VALUE - Long.MIN_VALUE, the beginning of time...
-      bb.putLong(0xffffffffffffffffL);
-      endAtMinlong = true;
-    }
+    boolean endAtMinlong = Long.MIN_VALUE == then;
```

Comment:
Not sure `endAtMinlong` is still needed because in the case of `preBoundaryScan`, both keys should be 0xFFFFFFFFFFFFFFFFL plus a 0x0 byte, thus fetching nothing.

Reply:
The use of endAtMinLong avoids a larger allocation if it is true, so let's keep it.

Reply:
Agreed, but if you care about allocations, you should always add a trailing 0x00 for endkey instead of copying it each time ;).
Also, in the case of preBoundary and endAtMinlong you could `scan.setStopRow(endkey);` to avoid a new allocation. In the end, it doesn't really matter as this case should never happen because of previous tests.